paper/paper.md

```diff
@@ -0,0 +1,137 @@
+It has a permissive license and can be extended to support additional conversions from custom R classes.
+
+The package `rdata` has been designed as a pure Python package with minimal dependencies, so that it can be easily integrated inside other libraries and applications.
+It currently powers the functionality offered in the `scikit-datasets` package [@diaz-vico+ramos-carreno_2023_scikitdatasets] for loading datasets from the CRAN repository of R packages.
+This functionality is used for fetching the functional datasets provided in the `scikit-fda` library [@ramos-carreno+_2024_scikitfda], whose development was the main reason for the creation of the `rdata` package itself.
+
+# Features
+
+The package `rdata` is intended to be both flexible and easy to use.
+In order to be flexible, the parsing of the RData format and the conversion of the parsed structures to appropriate Python objects have been splitted in two steps.
+This allows advanced users to perform custom conversions without losing information.
+Most users, however, will want to use the default conversion routine, which attempts to convert data
+to a standard Python representation which preserves most part of the information.
+
+```python
+import rdata
+
+converted = rdata.read_rda("dataset.rda")
+converted
+```
+
+This is equivalent to the following code, in which the two steps have been performed separatedly.
+
+```python
+import rdata
+
+parsed = rdata.parser.parse_file("dataset.rda")
+converted = rdata.conversion.convert(parsed)
+```
+
+The function `parse_file()` of the parser module is used to parse the RData file, returning a tree-like structure of Python objects that contains a representation of the basic R objects conforming the dataset.
+The function `convert()` of the conversion module transforms that representation to the final Python objects, such as lists, dictionaries or dataframes, that users can manipulate.
+
+Advanced users will probably require loading datasets which contain non standard S3 or S4 classes, translating each of them to a custom Python class.
+This is easy to achieve using `rdata` by simply creating a constructor function that receives the converted object representation and its attributes, and returns a Python object of the desired type.
+As an example, consider the following simple code that constructs a `Pandas` [@pandasdevelopmentteam_2024_pandasdev] `Categorical` object from the internal representation of an R `factor`.
+
+```python
+import pandas
+
+
+def factor_constructor(obj, attrs):
+    values = [attrs['levels'][i - 1] if i >= 0 else None for i in obj]
+
+    return pandas.Categorical(values, attrs['levels'], ordered=False)
```

Comment:
```suggestion
import pandas

def factor_constructor(obj, attrs):
    values = [attrs['levels'][i - 1] if i >= 0 else None for i in obj]
    return pandas.Categorical(values, attrs['levels'], ordered=False)
```

I'd suggest to remove these two empty lines for better readability in the rendered pdf.